src/data_types/indexed_tensor.hpp

```diff
@@ -152,6 +152,39 @@ KOKKOS_FUNCTION void internal_tensor_mul(
     ((internal_tensor_mul_elem<GlobalTensorIndexIdMap, Is>(result, t...)), ...);
 }
 
+template <char ID, class AllIndexIdMaps>
+KOKKOS_INLINE_FUNCTION void check_id_validity()
+{
+    constexpr std::size_t n_occurrences
+            = char_occurrences_v<ID, index_identifiers_t<AllIndexIdMaps>>;
+    if constexpr (n_occurrences == 2) {
+        using RelevantVectorIndexSets = relevant_vector_index_sets_t<ID, AllIndexIdMaps>;
+        using Set1 = ddc::type_seq_element_t<0, RelevantVectorIndexSets>;
+        using Set2 = ddc::type_seq_element_t<1, RelevantVectorIndexSets>;
+        constexpr bool has_covariant_idx = (is_covariant_vector_index_set_v<Set1>)
+                                           || (is_covariant_vector_index_set_v<Set2>);
+        constexpr bool has_contravariant_idx = (is_contravariant_vector_index_set_v<Set1>)
+                                               || (is_contravariant_vector_index_set_v<Set2>);
```

Comment:
I hope my question makes sense, is the whole vector index set supposed to be fully covariant or contravariant ? The vector set cannot have a mix of covariant/contravariant indices ?

Reply:
The question makes sense. While it is theoretically possible to build a vector containing both contravariant and covariant components it makes no sense to do so. The covariant and contravariant indices represent different bases. So writing `DVector<R, Theta_cov>` makes as much sense as `DVector<R, X>`

Reply:
mmh are you saying that $a_i^j = b_{i, k} c^{k, j}$ does not make sense ?

Reply:
No. This is ok. I do this for the Jacobian. It means that the matrix is expressed using 2 vector index sets.
E.g 
$J_i^k=\frac{\partial \xi_i}{\partial \eta_j}$
With $\xi_i$ coming from the VectorIndexSet {x,y} and $\eta_k$ coming from the VectorIndexSet {r,theta} (derivatives lead to inverted indices this is actually $`{\partial_\eta}^k`$ from the VectorIndexSet $`\{\partial^r, \partial^\theta\}`$ in tensor notation)

Reply:
The row indices are associated with one basis while the column indices are associated with another. What does not make sense is having row indices that represent elements from a variety of bases

Reply:
$a_i^j$ should have a type like:
```cpp
DTensor<VectorIndexSet<R, Theta>, VectorIndexSet<R_cov, Theta_cov>>
```
Each VectorIndexSet is either covariant or contravariant but the matrix has one index from each set

Reply:
Oh I see thanks